src/middlewared/middlewared/plugins/update_/freebsd_to_scale_linux.py

```diff
@@ -9,14 +9,20 @@
 
 
 class UpdateService(Service):
+
+    @private
+    def remove_files(self):
+        with contextlib.suppress(FileNotFoundError):
+            for i in ("/data/freebsd-to-scale-update", "/var/lib/dbus/machine-id", "/etc/machine-id"):
+                os.unlink(i)
+
     @private
     @job()
     async def freebsd_to_scale(self, job):
-        logger.info("Updating FreeBSD installation to SCALE")
-
-        with contextlib.suppress(FileNotFoundError):
-            os.unlink("/data/freebsd-to-scale-update")
+        logger.info("Updating CORE installation to SCALE")
 
+        await self.middleware.run_in_thread(self.remove_files)
+        await run(["systemd-machine-id-setup"], check=False)
```

Comment:
Will it be a no-op if machine-id exists already (if we upgraded using ISO instller which can set up machine id)

Reply:
Yes, `systemd-machine-id-setup` doesn't change an existing `/etc/machine-id` so it will be a no-op.